### PR TITLE
Payment and paid event combined

### PR DESCRIPTION
**Pull request by Nesssu**
The payment and paid event branches are now combined here and everything works at least for me :)

### CODE REVIEW

File: client/src/components/EventItem.js [hunk 1]
```diff
@@ -137,11 +161,45 @@ function EventItem(props) {
...
-    props.toggleUpdateEvents();
+  const handlePayment = async (e) => {
+    e.preventDefault()
+
+    setOpenPayment(false);
+
+    const data = {
+      userId: props.currentUser._id,
+      eventId: props.event._id
+    };
```
Comment (Enlarem) [inline client/src/components/EventItem.js:177]: Here, the userId is undefined when I test it on my computer. So it makes the payment process not working

**Review by Enlarem — APPROVED** [review]
Everything seem be working from my part !

File: server/routes/api.js [hunk 2]
```diff
@@ -16,13 +17,13 @@ const upload = multer({storage});
...
 
 
-require('../auth/passport')(passport)
+require('../auth/passport')(passport);
 router.use(passport.initialize());
 
 //finds all the members in the DB if authenticated
 router.get("/members/", passport.authenticate('jwt', {session: false}), async (req, res) => {
     try {
-      const members  = await Member.find({}).select("-password").sort({firstname: 1, lastname: 1});
```
Comment (jannejjj) [inline server/routes/api.js:25]: Wasn't this sorting here to show the members in alphabetical order? 🤔

File: server/routes/api.js [hunk 3]
```diff
@@ -146,53 +146,28 @@ router.post('/register',
       
 });
 
-// Update the role and admin permissions for the member
```
Comment (jannejjj) [inline server/routes/api.js:149]: I think everything else is good, but we seem to have lost the role/admin/delete member handling in the back-end

File: server/routes/api.js [hunk 4]
```diff
@@ -146,53 +146,28 @@ router.post('/register',
...
+    let paid = {};
 
-    // Find the IDs of the events that the user is participating in
+    // Find the IDs of the events that the user is interested in or has paid for
     await Member_event.find({member: id})
     .then((docs) =>
+    {
+        docs.forEach(item => {
+            eventIDs.push(item.event);
+            paid[item.event] = item.paid;
```
Comment (jannejjj) [inline server/routes/api.js:164]: member_event no longer has a `paid` field (we now have a ticket entity for that), so this (and other places) should be changed to not use it.  Whether or not a user has paid for an event should be found out by checking if they have a ticket for it :D

**Comment by Nesssu** [comment]
Now everything should be fixed again

**Review by jannejjj — APPROVED** [review]
With the newest changes, everything looks good and works fine!

**Review by mh-lut — APPROVED** [review]

**Pull request merged**